README.md

```diff
@@ -4,65 +4,85 @@
+
+1.  Compile in your logal Go environment,
+2.  Build using the packaged Gradle scripts (including the 'gogradle' plugin)
+
+## Build the binary with Go
 
-Make sure that you have Go installed [installing Go](https://golang.org/doc/install), and `$GOPATH` is defined [Go development environment](https://golang.org/doc/code.html).
+Make sure that you have Go installed [installing
+Go](https://golang.org/doc/install), and `$GOPATH` is defined [Go development
+environment](https://golang.org/doc/code.html).
 
-Then download the source code of the OpenWhisk CLI and the dependencies by typing:
+Then download the source code of the OpenWhisk CLI and the dependencies by
+typing:
 
 ```
 $ cd $GOPATH
 $ go get github.com/apache/incubator-openwhisk-cli
 ```
 
-
-## Build the binary with Go
-
-Open an terminal, go to the directory of OpenWhisk CLI home directory, and build the binary via the following command:
+Open an terminal, go to the directory of OpenWhisk CLI home directory, and build
+the binary via the following command:
 
 ```
 $ go build -o wsk
 ```
 
-If you would like to build the binary for a specific operating system, you may add the arguments GOOS and
-GOARCH into the Go build command. Since it is only applicable under amd64 architecture, you have to set GOARCH
-to amd64. GOOS can be set to "linux" "darwin" or "windows".
+If you would like to build the binary for a specific operating system, you may
+add the arguments GOOS and GOARCH into the Go build command. Since it is only
+applicable under amd64 architecture, you have to set GOARCH to amd64. GOOS can
+be set to "linux" "darwin" or "windows".
 
 For example, run the following command to build the binary for Linux:
 
 ```
 $ GOOS=linux GOARCH=amd64 go build -o wsk
 ```
 
-If it is executed successfully, you can find your binary `wsk` directly under OpenWhisk CLI home directory.
+If it is executed successfully, you can find your binary `wsk` directly under
+OpenWhisk CLI home directory.
 
-## Build the binary with Docker and Gradle
+## Build the binary with Gradle
 
-This is the second choice for you to build the binary. Make sure that you have Docker and gradle on your machine:
-[installing Docker](https://docs.docker.com/engine/installation/) and [installing Gradle](https://gradle.org/install) for your local machine.
+**Note:** For those who may have used the Gradle build previously, it has been
+re-engineered to no longer required Docker or Go to be pre-installed on your
+system.  Using the [gogradle](https://github.com/gogradle/gogradle) plugin, the
+script now uses a go environment if you have it, but will also create a local
+Go environment if you do now.
 
-After that, open an terminal, go to the directory of OpenWhisk CLI home directory, and
-build the binary via the following command under Linux or Mac:
+To build with Gradle, open an terminal, go to the directory of OpenWhisk CLI
+home directory, and build the binary via the following command under Linux or
+Mac:
```

Comment:
Can we add Windows to this now?

Reply:
Line 88 and 94 should be `distDocker` instead of `buildBinaries`.

Reply:
Windows command is two lines down from this (.\gradlew.bat).